client/client_update.go

```diff
@@ -78,7 +79,17 @@ func (u *UpdateClient) getUpdateInfo(api ApiRequester, process RequestProcessing
 
 	defer r.Body.Close()
 
+	respdata, err := ioutil.ReadAll(r.Body)
```

Comment:
Why do we need whole this section now?

Reply:
Because we can only read from the body once.

Reply:
It is processed once in one of the process functions. And then again in NewApiError, if there is an error.